python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -524,16 +526,14 @@ def asset_partition_key_for_output(self, output_name: str = "result") -> str:
         return self.partition_key
 
     def has_tag(self, key: str) -> bool:
-        per_invocation_properties = self._check_bound_to_invocation(
-            fn_name="has_tag", fn_type="method"
-        )
-        return key in per_invocation_properties.tags
+        return key in self._run_tags
```

Comment:
it looks like the `_pending_invocation_properties.tags` also includes tags on the op definition https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/execution/context/invocation.py 

Only having the run tags seems in line with the behavior of the real OpExecutionContext, but just want to confirm that removing access to the op def tags is intentional

Reply:
Ah yes - I should have called that out explicitly, but that is intentional. I think the current behavior is unexpected.